runtime/autoload/provider/pythonx.vim

```diff
@@ -13,25 +13,36 @@ function! provider#pythonx#Detect(ver) abort
   let skip = exists(skip_var) ? {skip_var} : 0
   if exists(host_var)
     " Disable auto detection
-    let [check, err] = s:check_interpreter({host_var}, a:ver, skip)
-    return check ? [{host_var}, err] : ['', err]
+    let [chk, err] = s:check_interpreter({host_var}, a:ver, skip)
+    return chk ? [{host_var}, err] : ['', err]
   endif
 
   let detect_versions = (a:ver == 2) ?
-        \   ['2.7', '2.6', '2', '']
-        \ : ['3.5', '3.4', '3.3', '3.2', '3', '']
+        \   ['2', '']
+        \ : ['3', '']
 
+  let full_err = ''
   for prog in map(detect_versions, "'python' . v:val")
-    let [check, err] = s:check_interpreter(prog, a:ver, skip)
-    if check
-      let [check, err] = s:check_version(prog, a:ver, skip)
-      return [prog, err]
+    let [chk, err] = s:check_interpreter(prog, a:ver, skip)
+    if chk
+      let [chk, err] = s:check_version(prog, a:ver, skip)
+      if chk
+        return [prog, err]
+      endif
+    endif
+
+    " Accumulate errors in case we don't find
+    " any suitable Python interpreter.
+    let prog_err = prog . ': ' . err
+    if full_err == ''
+      let full_err = prog_err
+    else
+      let full_err = full_err . "\n" . prog_err
     endif
   endfor
 
-  " No Python interpreter
-  return ['', 'Neovim module installed Python'
-        \ .a:ver.' interpreter is not found.']
+  " No suitable Python interpreter found.
+  return ['', full_err]
```

Comment:
I'm not sure how to even look at the output of this (i.e. the contents of `full_err`), but the output of `echom` looked good.